tests/src/Kernel/Entity/EntityGroupsTest.php

```diff
@@ -0,0 +1,216 @@
+   */
+  protected $group2;
+
+  /**
+   * @var string
+   *
+   * The machine name of the group node type.
+   */
+  protected $groupBundle;
+
+  /**
+   * @var string
+   *
+   * The machine name of the group content node type.
+   */
+  protected $groupContentBundle;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installConfig(['og']);
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('entity_test');
+    $this->installSchema('system', 'sequences');
+
+    $this->groupBundle = Unicode::strtolower($this->randomMachineName());
+    $this->groupContentBundle = Unicode::strtolower($this->randomMachineName());
+
+    // Create users.
+    $this->user1 = User::create(['name' => $this->randomString()]);
+    $this->user1->save();
+
+    $this->user2 = User::create(['name' => $this->randomString()]);
+    $this->user2->save();
+
+    $this->user3 = User::create(['name' => $this->randomString()]);
+    $this->user3->save();
+
+    // Define the group content as group.
+    Og::groupManager()->addGroup('entity_test', $this->groupBundle);
+
+    // Create a group and associate with user 1.
+    $this->group1 = EntityTest::create([
+      'type' => $this->groupBundle,
+      'name' => $this->randomString(),
+      'user_id' => $this->user1->id(),
+    ]);
+    $this->group1->save();
+
+    // Create a group and associate with user 2.
+    $this->group2 = EntityTest::create([
+      'type' => $this->groupBundle,
+      'name' => $this->randomString(),
+      'user_id' => $this->user2->id(),
+    ]);
+    $this->group2->save();
+  }
+
+  /**
+   * Tests owner groups are returned correctly.
+   */
+  public function testOwnerGroupsOnly() {
+    $actual = Og::getEntityGroups('user', $this->user1->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user1->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user2->id()));
+
+    $actual = Og::getEntityGroups('user', $this->user2->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user2->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user1->id()));
+  }
+
+  /**
+   * Tests other groups users are added to.
+   */
+  public function testOtherGroups() {
+    // Should be a part of no groups.
+    $this->assertEquals([], Og::getEntityGroups('user', $this->user3->id()));
```

Comment:
yeah, cleaning the arguments will result with nicer code (`Og::getEntityGroups($this->user3)`)